cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -3157,6 +3156,11 @@
 (rule (lower_icmp_bool (IcmpCondResult.Condition producer cc))
       (with_flags producer (x64_setcc cc)))
 
+;; Emit a conditional move based on the result of an icmp.
+(decl select_icmp (IcmpCondResult Value Value) ValueRegs)
```

Comment:
This might not be the most reusable signature, should I rework this to take the type as an argument and `ValueRegs` arguments instead of `Value`?

Reply:
That might be slightly more idiomatic for the "inst helpers" layer, though I don't think it's too important either way; strong types mean we can always refactor easily and relatively safely...